tests/ArmTemplates.Tests/Moqs/ApiClients/MockOpenIdConnectProviderClient.cs

```diff
@@ -0,0 +1,80 @@
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Moqs.ApiClients;
+using Moq;
+using Moq.Protected;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Moqs.IdentityProviderClients
+{
+    public class MockOpenIdConnectProviderClient
+    {
+        public const string TemplateType = "Microsoft.ApiManagement/service/identityProviders";
+        public const string ClientSecretDefaultValue = "clientSecretValue";
+        public static OpenIdConnectProviderProperties GetMockedOpenIdConnectProviderProperties(string providerName)
+        {
+            return new OpenIdConnectProviderProperties
+            {
+                ClientId = "clientid",
+                ClientSecret = null,
+                Description = "Description"
+            };
+        }
+
+        public static List<OpenIdConnectProviderResource> GenerateMockedIdentityProviderList(List<string> providerNames) 
+        {
+            var openIdConnectProviderList = new List<OpenIdConnectProviderResource>();
+            foreach(var providerName in providerNames)
+            {
+                var properties = GetMockedOpenIdConnectProviderProperties(providerName);
+                openIdConnectProviderList.Add(new OpenIdConnectProviderResource
+                {
+                    OriginalName = providerName,
+                    Name = providerName,
+                    Type = TemplateType,
+                    Properties = properties
+                });
+            }
+
+            return openIdConnectProviderList;
+        }
+
+        public static IOpenIdConnectProvidersClient GetMockedOpenIdConnectProviderClient(List<string> providerNames)
+        {
+            var mockServiceProviderApiClient = new Mock<IOpenIdConnectProvidersClient>(MockBehavior.Strict);
+
+            mockServiceProviderApiClient
+                .Setup(x => x.GetAllAsync(It.IsAny<ExtractorParameters>()))
+                .ReturnsAsync((ExtractorParameters _) => GenerateMockedIdentityProviderList(providerNames));
+
+            mockServiceProviderApiClient
+                .Setup(x => x.ListOpenIdConnectProviderSecrets(It.IsAny<string>(), It.IsAny<ExtractorParameters>()))
+                .ReturnsAsync((string _, ExtractorParameters _) => new OpenIdConnectProviderSecret() 
+                {
+                    ClientSecret = ClientSecretDefaultValue
+                });
+
+            return mockServiceProviderApiClient.Object;
+        }
+
+        public static async Task<IOpenIdConnectProvidersClient> GetMockedHttpOpenIdConnectProviderClient(string responseFileLocation)
+        {
+            var mockedProcessor = new Mock<IOpenIdConnectProviderProcessor>(MockBehavior.Loose).Object;
+            var mockedClient = new Mock<OpenIdConnectProviderClient>(MockBehavior.Strict, await MockClientUtils.GenerateMockedIHttpClientFactoryWithResponse(responseFileLocation), mockedProcessor);
```

Comment:
After several PRs I would suggest to move mock of `AzureCliAuthenticator` to some base class, because these several lines are just being added to every PR.

I am talking about doing change in this PR, but let's do it in the next one